nucypher/blockchain/eth/clients.py

```diff
@@ -85,39 +84,32 @@ def _add_default_middleware(self):
         endpoint_uri = getattr(self.w3.provider, "endpoint_uri", "")
         if "infura" in endpoint_uri:
             self.log.debug("Adding Infura RPC retry middleware to client")
-            self.add_middleware(InfuraRetryRequestMiddleware)
+            self.add_middleware(InfuraRetryRequestMiddleware, name="infura_retry")
         elif "alchemyapi.io" in endpoint_uri:
             self.log.debug("Adding Alchemy RPC retry middleware to client")
-            self.add_middleware(AlchemyRetryRequestMiddleware)
+            self.add_middleware(AlchemyRetryRequestMiddleware, name="alchemy_retry")
         else:
             self.log.debug("Adding RPC retry middleware to client")
-            self.add_middleware(RetryRequestMiddleware)
-
-        # poa middleware
-        chain_id = self.chain_id
-        is_poa = chain_id in POA_CHAINS
-
-        self.log.debug(
-            f"Blockchain: {self.chain_name} (chain_id={chain_id}, poa={is_poa})"
+            self.add_middleware(RetryRequestMiddleware, name="retry")
+
+        # add POA middleware irrespective of chain
+        poa_middleware_name = "poa"
+        self.log.info("Injecting POA middleware at layer 0")
+        self.inject_middleware(
+            geth_poa_middleware,
+            layer=0,
+            name=poa_middleware_name,
         )
-        if is_poa:
-            # proof-of-authority blockchain
-            self.log.info("Injecting POA middleware at layer 0")
-            self.inject_middleware(geth_poa_middleware, layer=0, name="poa")
 
         # simple cache middleware
         self.log.debug("Adding simple_cache_middleware")
```

Comment:
Let's make all the logs in this method to be info level.

Reply:
✔️